google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -587,8 +596,12 @@ def _get_initial_request(self, stream_ack_deadline_seconds):
             stream_ack_deadline_seconds=stream_ack_deadline_seconds,
             subscription=self._subscription,
             client_id=self._client_id,
-            max_outstanding_messages=self._flow_control.max_messages,
-            max_outstanding_bytes=self._flow_control.max_bytes,
+            max_outstanding_messages=(
+                0 if self._use_legacy_flow_control else self._flow_control.max_messages
```

Comment:
I don't quite understand this. Are we disabling client-side flow control altogether if _use_legacy_flow_control is true?

Reply:
No, the default right now (and with this change) is to enforce flow control as the Cloud Pub/Sub server side and also at the client side (see `maybe_pause_consumer()`, `maybe_resume_consumer()` and `_maybe_release_messages()`).

When use_legacy_flow_control=True, only the client side flow control continues to be enforced, the Cloud Pub/Sub server will not enforce any flow control. (this is the state of the world today, we are enabling server side flow control --along with the current client side-- and want to give users a path to disable this functionality if it causes issues.)

Reply:
NVM, I had thought this was in the constructor, not when sending the request. This makes sense.